database/src/main/java/database/service/SqlDatabaseService.java

```diff
@@ -503,4 +588,96 @@ private <T extends BaseEntity> T createAndFillEntity(Class<? extends BaseEntity>
         }
         return entity;
     }
+
+    private <T extends BaseEntity> List<T> createAndFillEntities(Class<? extends BaseEntity> entityClass,
+                                                                 ResultSet resultSet)
+            throws ReflectiveOperationException, SQLException {
+        List<T> entities = new ArrayList<>();
+        while (resultSet.next()) {
+            entities.add(createAndFillEntity(entityClass, resultSet));
+        }
+        return entities;
+    }
+
+    private <T extends BaseEntity> void validateEntities(List<T> entities) {
+        if (entities == null || entities.isEmpty()) {
+            throw new IllegalArgumentException(ENTITIES_LIST_NULL_OR_EMPTY);
```

Comment:
Please use custom exception which extends from `BadRequestException` to return 400 error code instead of 500.